vizro-core/tests/unit/vizro/models/_components/form/test_date_picker.py

```diff
@@ -0,0 +1,171 @@
+        actions_chain = date_picker.actions[0]
+
+        assert actions_chain.trigger.component_property == "value"
+
+    @pytest.mark.parametrize("min, max", [("2024-01-01", None), (None, "2024-01-01"), ("2024-01-01", "2024-02-01")])
+    def test_valid_min_max(self, min, max):
+        date_picker = vm.DatePicker(min=min, max=max)
+
+        assert date_picker.min == (datetime.strptime(min, "%Y-%m-%d").date() if min else None)
+        assert date_picker.max == (datetime.strptime(max, "%Y-%m-%d").date() if max else None)
+
+    def test_validate_max_invalid(self):
+        with pytest.raises(
+            ValidationError, match="Maximum value of component is required to be larger than minimum value."
+        ):
+            vm.DatePicker(min="2024-02-01", max="2024-01-01")
+
+    @pytest.mark.parametrize(
+        "range, value",
+        [
+            (False, "2024-01-01"),
+            (False, date(2024, 1, 1)),
+            (True, ["2024-01-01", "2024-02-01"]),
+            (True, [date(2024, 1, 1), date(2024, 2, 1)]),
+        ],
+    )
+    def test_validate_datepicker_value_valid(self, range, value):
+        date_picker = vm.DatePicker(min="2024-01-01", max="2024-02-01", range=range, value=value)
+
+        value_to_date = pd.to_datetime(value).to_pydatetime()
+        if isinstance(value_to_date, Collection):
+            value_to_date = [v.date() for v in value_to_date]
+        else:
+            value_to_date = value_to_date.date()
+
+        assert date_picker.value == value_to_date
+
+    @pytest.mark.parametrize(
+        "range, value",
+        [
+            (False, "2024-01-01"),
+            (False, date(2024, 1, 1)),
+            (True, ["2024-01-01", "2024-02-01"]),
+            (True, [date(2024, 1, 1), date(2024, 2, 1)]),
+        ],
+    )
+    def test_validate_datepicker_value_invalid(self, range, value):
+        with pytest.raises(ValidationError, match="Please provide a valid value between the min and max value."):
+            vm.DatePicker(min="2023-01-01", max="2023-02-01", range=range, value=value)
+
+    @pytest.mark.parametrize(
+        "range, value",
+        [
+            (False, "2024-01-01"),
+            (False, date(2024, 1, 1)),
+            (True, ["2024-01-01", "2024-02-01"]),
+            (True, [date(2024, 1, 1), date(2024, 2, 1)]),
+        ],
+    )
+    def test_validate_datepicker_range_valid(self, range, value):
+        date_picker = vm.DatePicker(min="2024-01-01", max="2024-04-01", range=range, value=value)
+        assert date_picker.range == range
+
+    @pytest.mark.parametrize(
+        "range, value",
+        [
+            (True, "2024-01-01"),  # range True produces DateRangePicker, value needs to be list with 2 dates
+            (True, ["2024-01-01"]),  # range True produces DateRangePicker, value needs to be list with 2 dates
+            (False, ["2024-01-01"]),  # range False produces DatePicker, value needs to be single date
```

Comment:
I didn't discussed this, and I guess that the `range` dictates the rules. @huong-li-nguyen @nadijagraca correct me if I'm wrong. 

Yes, it's possible to cast the value `list[str]` to `str` if the `range` is `False` or to set the range depending on what is the propagated value format. However that would increase the complexity for us and potentially for the user to understand I guess.

Reply:
I think it's less for smth that the user should understand, just one more option that would not fail. As in when they choose `range=False` but provide a list, if the list is `len(list) == 1`, we could make it work as a nice feature for the user

Reply:
We could add an additional check in validate_range to accommodate Max's suggestion, but the main question is what combinations do we want to allow. 
If we allow `range=False` and `value=["2024-01-01"]`,  do we want to allow `range=True` and `value=["2024-01-01"]` and handle this case in the background as well?

I'm tilting a bit to keeping it simple and having users provide single date for when range=False and list of dates for range=True, but I'm okay with implementing the suggested changed if majority agrees.

Reply:
It's fine for me, was just a suggestions. In your other case I would say the failure is clearer, because there is information missing with the second date

Reply:
Agree with @nadijagraca - this is fine how it is, let's keep it simple in this case. This handling is consistent with how we handle e.g. `RadioItems` where `value` has to be strictly a single value rather than allowing `[value]`. And vice versa with `Checklist` if you provide `value = 1` we don't convert to `[1]` at the moment